machine/v1beta1/types_vsphereprovider.go

```diff
@@ -198,8 +198,34 @@ type VSphereDisk struct {
 	// +kubebuilder:validation:Maximum=16384
 	// +required
 	SizeGiB int32 `json:"sizeGiB"`
+	// provisioningMode is an optional field that specifies the provisioning type to be used by this vSphere data disk.
+	// Allowed values are Thin, Thick, and EagerlyZeroed.
+	// When set to Thin, the disk will be made using thin provisioning allocating the bare minimum space.
+	// When set to Thick, the full disk size will be allocated when disk is created.
+	// When set to EagerlyZeroed, the full disk size will be allocated and all bytes will be zeroed.
+	// If not set, the setting will be provided by the default storage policy.
```

Comment:
What is the "default storage policy"? How does a user find out what this will default to if they don't set it? Can this default change over time?

Reply:
its based on vsphere itself.  Another layer of complexity is if you create a template, you can also set a clone level policy that will override the vsphere default and will trickle to these data disks.

Reply:
This feels like something that would good to clearly state for the users that may be reading this description. I'm not sure on what would be the best wording, but maybe something along the lines of:
> If not set, vSphere itself will be responsible for choosing the storage policy based on its defaults or platform configurations in place that may override these defaults.

WDYT?

Reply:
I made a small adjustment.  I'm talking w/ team to see if they would change more to it.